runtime/oti/vmargs_core_api.h

```diff
@@ -74,6 +76,19 @@ newJavaVMArgInfo(J9JavaVMArgInfoList *vmArgumentsList, char *optString, uintptr_
 intptr_t
 addXOptionsFile(J9PortLibrary* portLib, const char *xOptionsfileArg, J9JavaVMArgInfoList *vmArgumentsList, uintptr_t verboseFlags);
 
+/*
+ * Add a -Xoptionsfile argument, plus contents of specified file.
+ * This allocates memory for the options string and contents of the file
+ * @param portLib port library
+ * @param xOptionsfileArg -Xoptionsfile=<path>
+ * @param vmArgumentsList current list of arguments
+ * @param verboseFlags set to VERBOSE_INIT for verbosity
+ * @param isDefault set to TRUE if using default optionfile
+ * @return 0 on success, negative value on failure
+ */
+intptr_t
+addXOptionsFileImpl(J9PortLibrary* portLib, const char *xOptionsfileArg, J9JavaVMArgInfoList *vmArgumentsList, uintptr_t verboseFlags, UDATA isDefault);
```

Comment:
Modify addXOptionsFile and parseOptionsBuffer directly instead of creating a wrapper function. It's harder to read this way in my opinion